pkg/testutils/kubernetes/builders/pod.go

```diff
@@ -66,6 +77,11 @@ func (b *podBuilder) WithLabels(labels map[string]string) PodBuilder {
 	return b
 }
 
+func (b *podBuilder) WithLabel(name string, value string) PodBuilder {
+	b.labels[name] = value
+	return b
+}
```

Comment:
As we discussed, my only concern would be to clarify somewhere the interaction between this and `WithLabels`.